src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -575,98 +659,155 @@ private void replaceEC2Servers() {
-        }
         // Download trip planner JAR.
         lines.add(String.format("mkdir -p %s", jarDir));
         lines.add(String.format("wget %s -O %s/%s.jar", s3JarUrl, jarDir, jarName));
         if (graphAlreadyBuilt) {
             lines.add("echo 'downloading graph from s3'");
-            // Download Graph from S3 and spin up trip planner.
-            lines.add(String.format("aws s3 --region us-east-1 cp %s %s/Graph.obj", getS3GraphPath(), routerDir));
+            // Download Graph from S3.
+            lines.add(String.format("aws s3 --region us-east-1 cp %s %s/%s ", getS3GraphURI(), routerDir, OTP_GRAPH_FILENAME));
         } else {
+            // Download data bundle from S3.
+            lines.add(String.format("aws s3 --region us-east-1 cp %s /tmp/bundle.zip", getS3BundleURI()));
+            // Determine if bundle download was successful.
+            lines.add("[ -f /tmp/bundle.zip ] && BUNDLE_STATUS='SUCCESS' || BUNDLE_STATUS='FAILURE'");
+            // Create and upload file with bundle status to notify Data Tools that download is complete.
+            lines.add(String.format("echo $BUNDLE_STATUS > /tmp/%s", BUNDLE_DOWNLOAD_COMPLETE_FILE));
+            lines.add(String.format("aws s3 --region us-east-1 cp /tmp/%s %s", BUNDLE_DOWNLOAD_COMPLETE_FILE, joinToS3FolderURI(BUNDLE_DOWNLOAD_COMPLETE_FILE)));
+            // Put unzipped bundle data into router directory.
+            lines.add(String.format("unzip /tmp/bundle.zip -d %s", routerDir));
+            // FIXME: Add ability to fetch custom bikeshare.xml file (CarFreeAtoZ)
+            if (false) {
+                lines.add(String.format("wget -O %s/bikeshare.xml ${config.bikeshareFeed}", routerDir));
+                lines.add(String.format("printf \"{\\n  bikeRentalFile: \"bikeshare.xml\"\\n}\" >> %s/build-config.json\"", routerDir));
+            }
             lines.add("echo 'starting graph build'");
             // Build the graph if Graph object (presumably this is the first instance to be started up).
             if (deployment.r5) lines.add(String.format("sudo -H -u ubuntu java -Xmx6G -jar %s/%s.jar point --build %s", jarDir, jarName, routerDir));
             else lines.add(String.format("sudo -H -u ubuntu java -jar %s/%s.jar --build %s > $BUILDLOGFILE 2>&1", jarDir, jarName, routerDir));
-            // Upload the graph to S3.
-            if (!deployment.r5) lines.add(String.format("aws s3 --region us-east-1 cp %s/Graph.obj %s", routerDir, getS3GraphPath()));
+            // Upload the graph and build log file to S3.
+            if (!deployment.r5) {
+                lines.add(String.format("aws s3 --region us-east-1 cp %s/%s %s ", routerDir, OTP_GRAPH_FILENAME, getS3GraphURI()));
+                String s3BuildLogPath = joinToS3FolderURI(getBuildLogFilename());
+                lines.add(String.format("aws s3 --region us-east-1 cp $BUILDLOGFILE %s ", s3BuildLogPath));
```

Comment:
It may make more sense to upload the logs first in case the graph wasn't built. If the graph wasn't built, the script might error out on a file not found error perhaps.

Reply:
Will that actually happen? I suppose I can make the change regardless, but I don't think the execution of any line is dependent on the previous line, is it?